src/main/java/frc/robot/Drive.java

```diff
@@ -0,0 +1,212 @@
+          leftFront.set(0);
+          rightFront.set(0);
+          leftBack.set(0);
+          rightBack.set(0);
+          eDriveDone = true;
+        }
+
+        double allEncoder = lFrontEncoder.getPosition() + rFrontEncoder.getPosition();
+    
+      }
+
+    //This method is called on during autonomous for turning
+    public void turnDrive(int timeOut, double turnDegrees, double speed) {
+        //set turnDegrees parameters for this method to negative to turn left
+        //timeOut parameter should be in milliseconds
+        double desiredAngle = currentAngle + turnDegrees;
+
+        //statement above calculates the position of the desired angle
+        //based on the robots current orientation
+        double timeStarted = System.currentTimeMillis();
+        //timeStarted is recorded for timeOut
+
+        double error = desiredAngle - gyro.getAngle();
+
+        System.out.println("Gyro Angle: " + gyro.getAngle());
+        System.out.println("Desired Angle: " + desiredAngle);
+        System.out.println("Error: " + error);
+        //error between desiredAngle and our current angle is established
+        if(error < -2 || error > 2) {
+          //While loop will continue as long as error is not inbetween the slack range of -3 to 3
+          error = desiredAngle - gyro.getAngle();
+          //This calculates at the start of every loop to determine which way the robot will turn
+          if(error > 0) {
+              //If error is positive, the robot turns right
+              leftFront.set(speed);
+              rightFront.set(-speed);
+              leftBack.set(speed);
+              rightBack.set(-speed);
+          }
+          else {
+              //If error is negative, the robot turns left
+              leftFront.set(-speed);
+              rightFront.set(speed);
+              leftBack.set(-speed);
+              rightBack.set(speed);
+          }
+        }
+        else {
+          leftFront.stopMotor();
+          rightFront.stopMotor();
+          leftBack.stopMotor();
+          rightBack.stopMotor();
+          turnDrive = true;
+        }
+
+        //Stop all motors
+    }
+
+    public void angleAlign() {
+      double speed = 0;
+      double k = 0.03;
+      double angleOffset = aprilTags.getHorizontalDegToTarget();
+      alignDone = false;
+
+      if (angleOffset < -0.5 || angleOffset > 0.5) {
+        speed = angleOffset * k;
+        if (Math.abs(speed) > 0.1) {
+          speed = Math.copySign(0.1, speed);
+        }
+        else if (Math.abs(speed) < 0.075) {
+          speed = Math.copySign(0.075, speed);
+        }
+        leftFront.set(speed);
+        rightFront.set(-speed);
+        leftBack.set(speed);
+        rightBack.set(-speed);
+      }
+      else {
+        leftFront.stopMotor();
+        rightFront.stopMotor();
+        leftBack.stopMotor();
+        rightBack.stopMotor();
+        alignDone = true;
+      }
+    }
```

Comment:
I'd recommend moving this to Align.java since it looks like it's for horizontal distance movements?